subsys/bluetooth/host/l2cap.c

```diff
@@ -1475,7 +1475,11 @@ static void le_conn_req(struct bt_l2cap *l2cap, uint8_t ident,
 
 	LOG_DBG("psm 0x%02x scid 0x%04x mtu %u mps %u credits %u", psm, scid, mtu, mps, credits);
 
-	if (mtu < L2CAP_LE_MIN_MTU || mps < L2CAP_LE_MIN_MPS) {
+	/* Validate parameters. Requirements are from Core Spec v6.0, Vol 3.A.4.22. Valid credit
+	 * range is from 0 to UINT16_MAX, thus no credit validation is needed.
+	 */
+	if (!IN_RANGE(mtu, L2CAP_LE_MIN_MTU, BT_L2CAP_MAX_MTU) ||
```

Comment:
Previously I've actually seen Coverity complain about similar checks, i.e. checks where the code checks something that's given by the type. So since `BT_L2CAP_MAX_MTU` is UINT16_MAX and `mtu` us a `uint16_t`, we may actually see a Coverity issue for this (similar to checking if an unsigned integer is less than 0). 

You could use the same argument for MAX MTU as you do for credits; it's already implicitly checked by the type itself. 

I'll leave it up to you whether to keep the MAX_MTU check or not